services/common/src/components/permits/ReportPermitRequirementForm.tsx

```diff
@@ -21,108 +19,239 @@ import RenderRadioButtons from "@mds/common/components/forms/RenderRadioButtons"
+  console.log("HI TARA multipleConditions?", multipleConditions)
+  const getLinkedConditionList = () => {
+    if (!hasExistingRequirements || !selectedRequirement) {
+      return [];
+    }
+    const condition_ids = selectedRequirement.permit_condition_ids
+
+    const linked = condition_ids
+      .map((id) => {
+        return conditionMap[id];
+      }).sort((a, b) => a.stepPath.localeCompare(b.stepPath));
+
+    return <ul>
+      {linked.map((lc) => {
+        return <li key={lc.permit_condition_id}>{lc.stepPath}</li>
+      })}
+    </ul>;
+  };
 
   useEffect(() => {
     if (!canEditPermitConditions) {
       setIsEditMode(false);
     }
   }, [canEditPermitConditions]);
 
-  const handleDeleteReportRequirement = async ({ mine_report_permit_requirement_id }) => {
-    deleteConfirmWrapper("Report Requirement", async () => {
-      await dispatch(deleteMineReportPermitRequirement({ mineGuid, mine_report_permit_requirement_id })).then(async () => {
-        await refreshData();
-        setIsEditMode(false);
-      });
-    })
+  const handleDeleteReportRequirement = async ({ mine_report_permit_requirement_id, permit_condition_ids }) => {
+    const updateData = async () => {
+      await refreshData();
+      setIsEditMode(false);
+    };
+
+    const deleteFunction = async () => {
+      await dispatch(deleteMineReportPermitRequirement({ mineGuid, mine_report_permit_requirement_id }));
+      updateData();
+    };
+
+    if (permit_condition_ids.length > 1) {
+      const modalContent = {
+        title: "Confirm Deletion",
+        content: <>Are you sure you want to delete this Report Requirement? It will also be deleted from the following conditions: {getLinkedConditionList()}</>,
+        okText: "Delete",
+        onOk: deleteFunction
+      };
+      Modal.confirm(modalContent);
+
+    } else {
+      deleteConfirmWrapper("Report Requirement", deleteFunction);
+    }
   };
 
+
   const handleEditReportRequirement = async (values) => {
-    await dispatch(updateMineReportPermitRequirement({ mineGuid, values })).then(async () => {
+    let resp;
+    if (values.mine_report_permit_requirement_id) {
+      resp = await dispatch(updateMineReportPermitRequirement({ mineGuid, values }));
+    } else {
+      resp = await dispatch(createMineReportPermitRequirement({ mineGuid, values }));
+    }
+    if (resp.payload) {
       await refreshData();
-      setIsEditMode(false);
+      // this is for an edge case where an existing report is edited and so it has a form on the page
+      // and it was not getting updated so not showing the correct conditions there.
+      if (condition && values.mine_report_permit_requirement_id) {
+        const mainFormName = `${FORM.ADD_REPORT_TO_PERMIT_CONDITION}-${values.mine_report_permit_requirement_id}`
+        dispatch(change(mainFormName, 'permit_condition_ids', resp.payload.permit_condition_ids))
```

Comment:
this was my bug that wouldn't die (it did die, eventually). I tried a bunch of other ways to deal with this and this was the only one that didn't create worse bugs.